src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java

```diff
@@ -270,6 +313,32 @@ public void testElevationDiscrerteManualResolution() throws Exception {
         assertXpathEvaluatesTo("0.0/3.0/2.0", "//Layer/Extent", dom);
     }
 
+    @Test
+    public void testElevationDiscreteManualResolutionWithEnd() throws Exception {
+        setupVectorDimensionWithEnd(
+                ResourceInfo.ELEVATION,
+                "startElevation",
+                "endElevation",
+                DimensionPresentation.DISCRETE_INTERVAL,
+                2.0,
+                UNITS,
+                UNIT_SYMBOL);
+
+        Document dom = dom(get("wms?request=getCapabilities&version=1.1.1"), false);
+        // print(dom);
+
+        // check dimension has been declared
+        assertXpathEvaluatesTo("1", "count(//Layer/Dimension)", dom);
+        assertXpathEvaluatesTo("elevation", "//Layer/Dimension/@name", dom);
+        assertXpathEvaluatesTo(UNITS, "//Layer/Dimension/@units", dom);
+        assertXpathEvaluatesTo(UNIT_SYMBOL, "//Layer/Dimension/@unitSymbol", dom);
+        // check we have the extent
+        assertXpathEvaluatesTo("1", "count(//Layer/Extent)", dom);
+        assertXpathEvaluatesTo("elevation", "//Layer/Extent/@name", dom);
+        assertXpathEvaluatesTo("1.0", "//Layer/Extent/@default", dom);
+        assertXpathEvaluatesTo("1.0/3.0/2.0", "//Layer/Extent", dom);
```

Comment:
Someone reading the capabilities would conclude that the specification is nonsensical, as 2 does not fit in the min/max value... but there is also no way to indicate that the dimension is based on intervals, rather than single values. Looks like something the spec did not foresee.... I searched in both WMS 1.1 and WMS 1.3 and found no indication of how to handle such case. It's probably a good idea to explain this situation in the GeoServer documentation.

Reply:
GeoServer lets administrators set pretty much any resolution they want as long as it is a number so whether or not the capabilities is showing something nonsensical is really up to the administrator even without this change.